src/runtime/typemanager.cs

```diff
@@ -172,7 +172,11 @@ internal static IntPtr CreateType(ManagedType impl, Type clrType)
             // that the type of the new type must PyType_Type at the time we
             // call this, else PyType_Ready will skip some slot initialization.
 
-            Runtime.PyType_Ready(type);
+            if (Runtime.PyType_Ready(type) < 0)
+            {
+                throw new PythonException();
+            }
+            OperatorMethod.FixupSlots(type, clrType);
```

Comment:
Should not the fixup happen before PyType_Ready?

Reply:
It could be.

Reply:
@amos402 From [Python doc](https://docs.python.org/3.5/c-api/type.html#c.PyType_Ready): Finalize a type object. This should be called on all type objects to finish their initialization. This function is responsible for adding inherited slots from a type’s base class. Return 0 on success, or return -1 and sets an exception on error.

Unfortunately, the description does not clarify if the behavior could change in the future :/ I'd try to move `FixupSlots` up, and see if it works that way.